SimulationAgent.Test/DeviceConnection/DeviceConnectionActorTest.cs

```diff
@@ -34,7 +35,10 @@ public DeviceConnectionActorTest()
             this.scriptInterpreter = new Mock<IScriptInterpreter>();
             this.devices = new Mock<IDevices>();
             this.rateLimiting = new Mock<IRateLimiting>();
-            this.fetchLogic = new Mock<Fetch>(
+            this.credentialsSetupLogic = new Mock<CredentialsSetup>(
+                this.devices.Object,
+                this.logger.Object);
+            this.fetchLogic = new Mock<FetchFromRegistry>(
```

Comment:
Missing tests for credentialsSetup, register cases.

Reply:
I don't understand this comment, can you elaborate?

Reply:
I was referring to CredentialsSetup.cs, Register.cs, etc classes - unit tests for these classes are missing.

Reply:
We will add those tests as part of 'Unit Test Improvement' task.